unicorn-core/src/main/scala/org/virtuslab/unicorn/repositories/IdRepositories.scala

```diff
@@ -141,21 +141,19 @@ protected[unicorn] trait IdRepositories[Underlying] {
      * Hook executed after element is saved - if you want to do some stuff then, override it.
      *
      * @param elem element to save
-     * @param session implicit session
      */
-    protected def afterSave(elem: Entity)(implicit session: Session): Unit = {}
+    protected def afterSave(elem: Entity): Unit = {}
 
     /**
      * Saves multiple elements.
      *
      * @param elems elements to save
-     * @param session implicit database session
      * @return Sequence of ids
      */
-    def saveAll(elems: Seq[Entity])(implicit session: Session): Future[Seq[Id]] = session.withTransaction {
-      // conversion is required to force lazy collections
-      Future.sequence(elems.toIndexedSeq map save)
+    def saveAll(elems: Seq[Entity])(implicit ec: ExecutionContext): DBIO[Seq[Id]] = {
+      DBIO.sequence(elems.map(save))
```

Comment:
The collection itself may be lazy, e.g. after doing `map(f)`, the `f` is not applied until you actually try to retrieve an element (it may not be applied at all, if it happens that you don't need any elements). The `toIndexedSeq` call provides a solution.